modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/IgniteStatisticsRepository.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.ignite.internal.processors.query.stat;
+
+import org.apache.ignite.internal.processors.cache.query.QueryTable;
+
+import java.util.Collection;
+
+/**
+ * Repository to store all necessary statistics. Can request absent ones from cluster and store to
+ * {@link SqlStatisticsStoreImpl}.
+ */
+public interface IgniteStatisticsRepository {
+
+    /**
+     * Replace all table statistics with specified ones.
+     *
+     * @param tbl table.
+     * @param statistics collection of tables partition statistics.
+     * @param fullStat if {@code True} - replace whole statistics, try to merge with existing - otherwise.
+     */
+    void saveLocalPartitionsStatistics(QueryTable tbl, Collection<ObjectPartitionStatisticsImpl> statistics, boolean fullStat);
+
+    /**
+     * Get local partition statistics by specified table.
+     *
+     * @param tbl table to get statistics by.
+     * @return collection of partitions statistics.
+     */
+    Collection<ObjectPartitionStatisticsImpl> getLocalPartitionsStatistics(QueryTable tbl);
```

Comment:
I think it is not good idea to use QueryTable here, because in order to get stats for index you have to create QueryTable with name from index, and this could be a quite confusing.